Makes users with a "Floorlength Braid" trip occasionally

Rebruiza opened the pull request:
This _only_ affects the "Floorlength Braid" hairstyle.
This also converts hair flags to bitflags.

**Changelog**
:cl: Rebruiza
rscadd: Individuals with the "Floorlength Braid" hairstyle now occasionally trip on their hair.
/:cl:

PsiOmegaDelta commented:
I said it from the beginning, if this is to be a thing at all it should checking some sort of flag on the hairstyle itself, not checking for a specific hairstyle.

Rebruiza commented:
@PsiOmegaDelta No idea how to do that, though.

mloc commented:
:-1: hacky and pointless

ghost commented:
@PsiOmegaDelta I'll look into it.

Rebruiza commented:
PR closed for now.

ghost commented:
i have no idea how to port and i have no idea how to code :woman_shrugging:

Zero-Bits commented:
Oh mah gawd, did I try!

mustafakalash commented:
Make `h_style` store the hair's datum rather than its string. Add `var/trip_chance = 0` to the hair datums and give it a value for FLB. In `Move()`, check `if(prob(h_style.trip_chance)) trip()`.

What you have here is pretty insane and does not even solve the original issue Psi had, which was that you have a check specific looking at `h_style == "Floorlength Braid"` rather than adding a generic flag any hair style could have.

AtomicWorm commented:
You're doing the Lord's work. Please do carry on, bless you.

Hubblenaut commented:
If FLBs are that hated upon that they are going to be a major disadvantage and annoyance in game, just get rid of them instead.

AtomicWorm commented:
It's been done before, but somebody keeps adding it back.

dryerlint commented:
> If FLBs are that hated upon that they are going to be a major disadvantage and annoyance in game, just get rid of them instead.

This is funnier, though.

Lohikar commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -48,6 +48,8 @@
 	make_blood()
 
 /mob/living/carbon/human/Destroy()
+	GLOB.hair_trip_callback = null
```
This, uh, this isn't a global, at least in the aurora implementation of it.
Nor is movement_triggers.

BlueNexus commented:
I'm not going to allow this, as it actively penalises players for choosing a *hairstyle of all things*, without warning them beforehand. How would you like having to change your character, because you kept tripping everywhere? This is just a jab at people for using a hairstyle you don't personally like.

Either remove it or leave it be.

Nerev4r commented:
Lame

mustafakalash commented:
Fix the code and get a poll.

BlueNexus commented:
Something something it's not just about code quality

atlantiscze commented:
I share @BlueNexus 's opinion on this topic. Penalizing player for choosing a particular hair style by making them trip is just a giant :-1: thing to do.

If this hairstyle needs a change like this that much, just remove the hair style altogether.

mustafakalash commented:
Poll is well in favor.

BlueNexus commented on code/modules/mob/living/carbon/human/human_movement.dm:
```diff
@@ -106,3 +106,13 @@
+			for (var/thing in movement_triggers)
+				if (prob(movement_triggers[thing]))
+					var/datum/callback/cb = thing
+					cb.InvokeAsync()
```
..DM has async?
I can't find a single use of that in our _entire_ repo.

PsiOmegaDelta replied:
In this particular case I'm fairly sure Bay uses the moved event instead.

BlueNexus commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -48,6 +48,8 @@
 
 /mob/living/carbon/human/Destroy()
+	GLOB.hair_trip_callback = null
+	GLOB.movement_triggers = null
```
Neither of these variables should be global, as far as I can tell.

meyar commented:
As Mloc, Bluenexus and atlantiscze have all already pointed out, 👎

Are "petty/vindictive" and "meme" valid tags yet.

mustafakalash commented:
Non-code discussion belongs elsewhere. I understand that there are people who don't like it. Those people turned out to be the minority.

@Rebruiza In case you lost it in the swarm, here is my advice on moving forward: https://github.com/Baystation12/Baystation12/pull/19088#issuecomment-337679873 If you need help, let me know on Discord. If you decide it's not worth it, close the PR.

Hubblenaut commented:
>Poll is well in favor.

The poll representates the feeling of the entire server population on this topic, not however the actual set of people that would be negatively effected under it. The vote is entirely unrepresentative and can barely be taken as an argument or reason to bring a change to a game that will be nothing but an unnecessary negative effect to the minority that plays FBP characters.

How anyone could seriously claim that a vote that is all about putting a minority disadvantage for the sake or hilarity of it is beyond me.

Techhead0 commented:
>Non-code discussion belongs elsewhere.

I have nuked a recent comment in violation of this. If a thread exists, post your non-code opinions there.

EDIT: I've since been informed that the thread is currently locked. Regardless, please keep the bickering out of the PR so it can remain open for people with code comments.

PurpleMartinJCK commented:
The code for this is gross. I'm going to fix it.

Rebruiza commented:
Current commits are out of date while I attempt to make it properly locally.

PurpleMartinJCK commented:
#19187 for alt implimentation

mustafakalash commented:
Closed in favor of above.

PurpleMartinJCK commented:
Mine isn't the same as was voted on so this should stay open

mustafakalash commented:
Gem can reopen this if/when he pushes an actual implementation.

Rebruiza commented:
:+1:

mustafakalash commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -1517,4 +1523,12 @@
\ No newline at end of file
+	return getBrainLoss() >= maxHealth * 0.75
+
+/mob/living/carbon/human/proc/trip(source)
```
You're no longer using this proc.

mustafakalash commented on code/modules/mob/mob_movement.dm:
```diff
@@ -480,6 +480,9 @@
 
 #define DO_MOVE(this_dir) var/final_dir = turn(this_dir, -dir2angle(dir)); Move(get_step(mob, final_dir), final_dir);
 
+/mob/proc/check_slipmove()
```
You need to actually add this check to the movement proc, check how Steve did it.

mustafakalash replied:
handled.

mustafakalash commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -1250,6 +1250,12 @@
+	if(h_style)
+		var/datum/sprite_accessory/hair/S = hair_styles_list[h_style]
+		if(S && S.flags & HAIR_TRIPPABLE && prob(0.4))
+			slip(S, 4)
```
`return TRUE` under slip, `return FALSE` outside and after the if

mustafakalash replied:
handled

Techhead0 commented on code/modules/mob/living/carbon/human/human_defines.dm:
```diff
@@ -114,3 +114,6 @@
 	var/skin_state = SKIN_NORMAL
+
+	var/list/movement_triggers
+	var/datum/callback/hair_trip_callback
```
Dead vars.

mustafakalash replied:
Oops

ghost commented:
*sigh*

mloc commented:
:-1: blerp

Rebruiza commented:
y-you can't double-dislike! :cry:

Commit:
Makes users with a floorlength braid trip occasionally


Attempted code fix


Removes faulty GLOB's and changes ASync to moved_event


trip_chance


Final


wew


crash bandicoot funny meme


Removes dead variables
Removes stray indentation

KrabSpider commented:
We're HRP, and the hairstyle is usually hated because it's snowflakey/ruins immersion. I don't like it personally, but you don't see me pushing for something as petty as **tripping when running while having the hairstyle just because _you_ don't like the hairstyle.** Either remove it or keep it, don't choose a middle ground that worsens the experience of others because you want to punish them for having a preference you don't share.

chaoko99 commented:
It also breaks every regulation *ever*, both military and corporate, even the ones in real life. Frankly, I don't think this is *enough*. Getting your braid caught in shit should fucking scalp you.

HeyBanditoz commented:
That'd hurt like fuck. Paincrit.

On October 31, 2017 10:59:59 PM MDT, chaoko99 <notifications@github.com> wrote:
>It also breaks every regulation *ever*, both military and corporate,
>even the ones in real life. Frankly, I don't think this is *enough*.
>Getting your braid caught in shit should fucking scalp you.
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/Baystation12/Baystation12/pull/19088#issuecomment-340978564

BlueNexus commented:
This always has been, and still is, vindictive. We don't need to fuck around with punishing players for having a different hairstyle preference. It breaks every regulation ever? Then **remove it**. People pushing for this over the removal option are seriously vindictive, since they apparently just want to punish the people who use it.

PurpleMartinJCK commented:
@chaoko99 @KrabSpider  @BlueNexus 

The discussion has been had, there was a vote, there was lots of time to talk about it and the general discussion is now over. We're now just waiting on @xales to choose between this PR, the removal one, or neither. Please refrain from commenting on it unless you have code concerns. Thank you.

mustafakalash commented:
Okay. We've waited long enough without word. I'm just going to do what was voted in the poll.

BlueNexus commented:
You people are fucking malicious. Hope you're happy.

atlantiscze commented:
This is one of the situations in which i believe common sense should have taken precedence over a vote.

meyar commented:
Apparently Rule 1 wasn't clear enough an indicator. Good thing I'm retired.
